### PR TITLE
fix: section name for analytics

### PR DESCRIPTION
**Pull request by aronshamash**
- https://nidigitalsolutions.jira.com/browse/REPLAT-12182
- The section name we are sending to analytics is unreliable / incorrect because we use the `section` field from `Article` in TPA response.
- Until TPA can give us this data in a "nicer" way we have to look into the `tiles` array and find it there instead. This is how we do it for the `<title>` data anyway.
- This currently always "news" in the Scotland and Ireland section.

### Scotland inside current edition

Screenshot 2020-02-04 at 11 05 16

### Scotland outside current edition

Screenshot 2020-02-04 at 11 03 53

### CODE REVIEW

File: packages/ssr/src/lib/get-analytics-section.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  const titles = sections.map(section => section.title);
+
+  if (titles.length === 0) {
+    return null;
+  }
+
+  const nonNewsSection = titles.filter(title => title !== "news");
+
+  return nonNewsSection.length ? nonNewsSection[0].toLowerCase() : "news";
```
Comment (ValGeorgiev) [inline packages/ssr/src/lib/get-analytics-section.js]: can we have some unit tests for this file

Comment (aronshamash) [reply packages/ssr/src/lib/get-analytics-section.js]: well  I did think of that but `ssr` doesn't have any tests.

Comment (aronshamash) [reply packages/ssr/src/lib/get-analytics-section.js]: option  _maybe_ to put it in `utils` package ?

Comment (ValGeorgiev) [reply packages/ssr/src/lib/get-analytics-section.js]: yes, we can move it to utils package

Comment (aronshamash) [reply packages/ssr/src/lib/get-analytics-section.js]: I'll create unit tests there

File: packages/ssr/src/lib/get-analytics-section.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+module.exports = article => {
+  // Get the section for an article, preferring it not to be news
+
```
Comment (apostolnikov) [inline packages/ssr/src/lib/get-analytics-section.js]: exports { tiles } =>

File: packages/ssr/src/lib/get-analytics-section.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  const slices = tiles.reduce((acc, tile) => {
+    acc.push(...tile.slices);
+    return acc;
+  }, []);
+
+  const sections = slices.reduce((acc, slice) => {
+    acc.push(...slice.sections);
+    return acc;
```
Comment (apostolnikov) [inline packages/ssr/src/lib/get-analytics-section.js]: this one and the one above could use implicit return 
const sections = slices.reduce((acc, slice) => acc.push(...slice.sections), []);

File: packages/ssr/src/lib/get-analytics-section.js (hunk 2 above)
Comment (aronshamash) [reply packages/ssr/src/lib/get-analytics-section.js]: done

File: packages/ssr/src/lib/get-analytics-section.js (hunk 3 above)
Comment (aronshamash) [reply packages/ssr/src/lib/get-analytics-section.js]: yes that seems good idea, but when I tried it tests all fail. Going to merge as it is as it is fairly urgent  fix

**Pull request merged**